src/Nethermind/Nethermind.Evm.Test/VirtualMachineTestsBase.cs

```diff
@@ -117,6 +118,18 @@ protected TestAllTracerWithOutput Execute(params byte[] code)
 
     protected virtual TestAllTracerWithOutput CreateTracer() => new();
 
+
+    protected T ExecuteBlock<T>(T tracer, byte[] code, ForkActivation? forkActivation = null) where T : IBlockTracer
+    {
+        (Block block, Transaction transaction) = PrepareTx(forkActivation ?? Activation, 100000, code);
+        tracer.StartNewBlockTrace(block);
+        ITxTracer txTracer = tracer.StartNewTxTrace(transaction);
+        _processor.Execute(transaction, block.Header, txTracer);
```

Comment:
Is there a risk that `_processor.Execute()` could throw and exception and the tracer never gets to execute `tracer.EndTxTrace()` and `tracer.EndBlockTrace()`?

Reply:
yes this is by desing (this is test)